scripts/macros/spells/spiritOfDeath.js

```diff
@@ -0,0 +1,221 @@
+    updates.actor.system.attributes.hp = {
+        formula: hpFormula,
+        max: hpFormula,
+        value: hpFormula
+    };
+    let animation = itemUtils.getConfig(workflow.item, 'animation') ?? 'none';
+    await Summons.spawn(sourceActor, updates, workflow.item, workflow.token, {
+        duration: 3600,
+        range: 90,
+        animation,
+        initiativeType: 'follows',
+        additionalSummonVaeButtons: [multiAttackFeatureData, reapingScytheFeatureData, hauntCreatureFeatureData].map(i => {return {type: 'use', name: i.name, identifier: i.flags['chris-premades'].info.identifier};})
+    });
+}
+async function attack({workflow}) {
+    if (workflow.targets.size !== 1 || workflow.advantage) return;
+    let effect = effectUtils.getEffectByIdentifier(workflow.actor, 'spiritOfDeathHauntCreature');
+    if (!effect) {
+        workflow.aborted = true;   
+        return;
+    }
+    let {targets: validTargetUuids, formula} = effect.flags['chris-premades'].spiritOfDeathHauntCreature;
+    if (!validTargetUuids.includes(workflow.targets.first().document.uuid)) {
+        workflow.aborted = true;
+        return;
+    }
+    workflow.advantage = true;
+    workflow.attackAdvAttribution.add(genericUtils.translate('DND5E.Advantage') + ': ' + effect.name);
+}
+async function late({workflow}) {
+    let effect = effectUtils.getEffectByIdentifier(workflow.actor, 'summonedEffect');
+    let originItem = fromUuidSync(effect?.origin);
+    let sourceActor = originItem?.actor;
+    if(!sourceActor)
+    {
+        return;
+    }
+    let featureData = await compendiumUtils.getItemFromCompendium(constants.packs.summonFeatures, 'Haunt Creature: Haunt', {object: true, getDescription: true, translate: 'CHRISPREMADES.Macros.SpiritOfDeath.HauntCreatureHaunt', identifier: 'spiritOfDeathHauntCreatureHaunt', flatDC: sourceActor.system.attributes.spelldc});
+    if (!featureData) {
+        errors.missingPackItem();
+        return;
+    }
+    let targetEffectData = {
+        name: genericUtils.translate('CHRISPREMADES.Macros.SpiritOfDeath.Haunted'),
+        img: workflow.item.img,
+        origin: workflow.item.uuid,
+        flags: {
+            dae: {
+                "specialDuration": [
+                    "zeroHP"
+                ]
+            }
+        }
+    };
+    let casterEffectData = {
+        name: workflow.item.name,
+        img: workflow.item.img,
+        origin: workflow.item.uuid,
+        flags: {
+            'chris-premades': {
+                spiritOfDeathHauntCreature: {
+                    targets: Array.from(workflow.targets).map(i => i.document.uuid),
+                }
+            }
+        }
+    };
+    let casterEffect = await effectUtils.createEffect(workflow.actor, casterEffectData, {concentrationItem: workflow.item, strictlyInterdependent: true, identifier: 'spiritOfDeathHauntCreature'});
+    await genericUtils.setFlag(casterEffect, 'chris-premades', 'macros.combat', ['spiritOfDeathHauntCreatureHaunt']);
```

Comment:
Don't set the flag after creation. Can either set it directly in the effectData or use `effectUtils.addMacro(casterEffectData, 'combat', ['spiritOfDeathHauntCreatureHaunt']);` prior to creating the effect.